### DISCUSSION TITLE
ASAR error : Invalid package

### DISCUSSION THREAD

**Discussion by M-Barari** [prisma/prisma#9366] [Q&A]
I'm using electron and when I build my app it gives me this error:

![Screenshot 2021-09-21 193721](https://user-images.githubusercontent.com/63962449/134196895-7488adb3-87ee-413c-8249-f8f86cb4e447.png)

I changed `prisma-client-genarator` output to my extraResources folder and it has been inserted into `win-unpacked/resource`

![Uploading Screenshot 2021-09-21 194122.png…]()

but still get error. what should I do??

**Comment by sdnts** [comment]
Hey @M-Barari, have you already tried what I suggested here? https://github.com/prisma/prisma/discussions/5200#discussioncomment-295575

**Reply by M-Barari** [reply]
Hi @madebysid , Not exactly your solution, but I kinda managed that by including in `build.file` of electron-builder, but far more importantly I found out that I can't use it in client-side ! and as I couldn't find a way to manage my endpoints by prisma in main.js , I'm gonna use `ipc` event-bus . Do you happen to know anyway to send endpoints to prisma?? for example `$axios.$get('api/users/all')` be handled by Prisma in main.js . I could do it by nuxt serverMiddleware but It's not working in exe file.

**Reply by M-Barari** [reply]
@madebysid , can u plz help me! I've been stuck for two weeks already!!  I was creating an app with nuxt+electron+prisma . As I didn't found any way to define endpoints to my prisma handler and Prisma error that can't be run on client-side , I decided to use electron event bus `ipcMain` and `ipcRenderer` to communicate between main.js and renderer but that had errors too . long story short, I stablished the communication by using [Michalzaq12's boilerplate](https://github.com/michalzaq12/electron-nuxt) but now that I'm gonna use Prisma , I get these errors: 
**dev:** 

![Screenshot 2021-09-23 180458](https://user-images.githubusercontent.com/63962449/134527374-07eb7327-1198-413e-b885-290bce8b6b7a.png)

**build:**

![Screenshot 2021-09-23 180320](https://user-images.githubusercontent.com/63962449/134527036-dd7ea80b-5bfd-4709-be5d-4f5061c1f8ea.png)

**also on both this warning:** 

![Screenshot ](https://user-images.githubusercontent.com/63962449/134528259-21af8065-5574-449b-b836-7187fcc029e7.png)
This is my package.json

```json
{
  "name": "mi",
  "version": "0.0.1",
  "description": "An electron-nuxt project",
  "main": "./dist/main/index.js",
  "scripts": {
    "dev": "node .electron-nuxt/dev.js",
    "build": "node .electron-nuxt/build.js",
    "test": "",
    "preinstall": "node .electron-nuxt/check-engines.js",
    "postinstall": "electron-builder install-app-deps "
  },
  "keywords": [],
  "author": "Mojtaba Barari <m.barari.sh@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/michalzaq12/electron-nuxt/blob/master/README.md",
  "engines": {
    "node": ">=12.13.0"
  },
  "resolutions": {
    "@types/node": "^14.6.2"
  },
  "devDependencies": {
    "@xpda-dev/core": "^0.1.6",
    "@xpda-dev/electron-builder": "^0.2.1",
    "@xpda-dev/electron-launcher": "^0.3.0",
    "@xpda-dev/webpack-step": "^0.1.2",
    "browser-env": "3.3.0",
    "cross-env": "7.0.3",
    "deepmerge": "4.2.2",
    "electron": "12.0.6",
    "electron-builder": "22.11.3",
    "electron-devtools-installer": "3.2.0",
    "nuxt": "2.15.6",
    "@nuxtjs/axios": "^5.13.6",
    "webpack": "4.44.2",
    "webpack-node-externals": "2.5.2",
    "express": "^4.17.1"
  },
  "dependencies": {
    "prisma": "^3.1.1",
    "@prisma/client": "^3.1.1"
  }
}
```

and electron-builder config:

```
const ICONS_DIR = 'build/icons/'

const windowsOS = {
  win: {
    icon: ICONS_DIR + 'win-icon.ico',
    publisherName: 'michal',
    target: 'nsis'
  },

  nsis: {
    differentialPackage: true
  }
}

const linuxOS = {
  linux: {
    icon: ICONS_DIR,
    target: 'deb'
  }
}

const macOS = {
  mac: {
    target: 'dmg',
    icon: ICONS_DIR + 'con.icns'
  },
  dmg: {
    contents: [
      {
        x: 410,
        y: 150,
        type: 'link',
        path: '/Applications'
      },
      {
        x: 130,
        y: 150,
        type: 'file'
      }
    ]
  }
}

module.exports = {
  productName: 'mi',
  appId: 'com.example.app',
  artifactName: 'setup-${version}.${ext}',
  directories: {
    output: 'build'
  },
  // default files: https://www.electron.build/configuration/contents
  files: [
    'package.json',
    {
      from: 'dist/main/',
      to: 'dist/main/'
    },
    {
      from: 'dist/renderer',
      to: 'dist/renderer/'
    },
    'src/extraResources/database.db',
    'node_modules/.prisma/**',
    'node_modules/@prisma/client/**'
  ],
  extraResources: [
    {
      from: 'src/extraResources/',
      to: ''
    }
  ],
  ...windowsOS,
  ...linuxOS,
  ...macOS
}
```

I dont know what to do anymore !!!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
